src/test/java/seedu/address/logic/parser/EditCommandParserTest.java

```diff
@@ -1,211 +1,536 @@
+                VALID_ITEM_INDEX + ITEM_TYPE_PROJECT + PREFIXED_NAME_ORBITAL + PREFIXED_TIME_ORBITAL
+                        + PREFIXED_WEBSITE_ORBITAL + PREFIXED_DESCRIPTION_ORBITAL,
+                new EditProjectCommand(INDEX_FIRST_ITEM, editProjectDescriptorNoTag));
+
+        // no project name, no project website
+        EditProjectDescriptor editProjectDescriptorNoNameNoTag = new EditProjectDescriptorBuilder()
+                .withTime(VALID_TIME_1)
+                .withDescription(VALID_DESCRIPTION_ORBITAL)
+                .withTags(VALID_TAG_BACKEND)
+                .build();
+
+        assertParseSuccess(parser,
+                VALID_ITEM_INDEX + ITEM_TYPE_PROJECT + PREFIXED_TIME_ORBITAL
+                        + PREFIXED_DESCRIPTION_ORBITAL + PREFIXED_TAG_BACKEND,
+                new EditProjectCommand(INDEX_FIRST_ITEM, editProjectDescriptorNoNameNoTag));
+
+
+        // no resume tags
+        EditResumeDescriptor editResumeDescriptorNoTag = new EditResumeDescriptorBuilder()
+                .withName(VALID_RESUME_NAME_ME)
+                .build();
+
+        assertParseSuccess(parser,
+                VALID_ITEM_INDEX + ITEM_TYPE_RESUME + PREFIXED_NAME_ME,
+                new EditResumeCommand(INDEX_FIRST_ITEM, editResumeDescriptorNoTag));
+
+        // no skill tags
+        EditSkillDescriptor editSkillDescriptorNoTag = new EditSkillDescriptorBuilder()
+                .withName(VALID_SKILL_NAME_REACT)
+                .withLevel(Level.BASIC)
+                .build();
+
+        assertParseSuccess(parser,
+                VALID_ITEM_INDEX + ITEM_TYPE_SKILL + PREFIXED_NAME_REACT + PREFIXED_BASIC,
+                new EditSkillCommand(INDEX_FIRST_ITEM, editSkillDescriptorNoTag));
+
+        // no skill name, no skill level
+        EditSkillDescriptor editSkillDescriptorNoNameNoLevel = new EditSkillDescriptorBuilder()
+                .withTags(VALID_TAG_FRONTEND, VALID_TAG_BACKEND)
+                .build();
+
+        assertParseSuccess(parser,
+                VALID_ITEM_INDEX + ITEM_TYPE_SKILL + PREFIXED_TAG_FRONTEND + PREFIXED_TAG_BACKEND,
+                new EditSkillCommand(INDEX_FIRST_ITEM, editSkillDescriptorNoNameNoLevel));
+    }
+
+    @Test
+    public void parse_compulsoryItemTypeMissing_failure() {
+        // missing item prefix
+        assertParseFailure(parser,
+                VALID_ITEM_INDEX + PREFIXED_NAME_GOOGLE + PREFIXED_ROLE_FRONTEND
+                        + PREFIXED_TIME_FROM + PREFIXED_TIME_TO + PREFIXED_INTERNSHIP_DESCRIPTION
+                        + PREFIXED_TAG_FRONTEND,
+                Item.MESSAGE_CONSTRAINTS);
+
+        // invalid item prefix
+        assertParseFailure(parser,
+                VALID_ITEM_INDEX + INVALID_TYPE_DESC + PREFIXED_NAME_GOOGLE + PREFIXED_ROLE_FRONTEND
+                        + PREFIXED_TIME_FROM + PREFIXED_TIME_TO + PREFIXED_INTERNSHIP_DESCRIPTION
+                        + PREFIXED_TAG_FRONTEND,
+                "The item type is invalid!");
```

Comment:
I might have changed this in another PR. Let me go find later..

Reply:
Right, I made this change in #283 (in Item.java [here](https://github.com/chrisjwelly/main/blob/375547802ff46e85c6a4b3714dde8113103f5b15/src/main/java/seedu/address/model/item/Item.java#L22), to be exact).

So you have the exact same message, but can you change it to `Item.MESSAGE_INVALID_ITEM_TYPE` instead?

Reply:
Thanks I have changed it!